### ISSUE TITLE
Strip referer to strict-origin-when-cross-origin in all cases

### ISSUE THREAD

**Issue by pes10k** [brave/brave-browser#13464]
This is a follow up to https://github.com/brave/brave-browser/issues/13434 (which is no longer needed)

Brave currently completely strips the referrer on all cross origin top-frame navigations.  This causes Brave to look like bots / fraud for some non-malicious systems (e.g., DDG and similar). 

We should change Brave's referrer policy to cap at (i.e. send less when the page requests less, but cap and default to `strict-origin-when-cross-origin`)

**Comment by kjozwiak** [comment]
Moving into `1.21.x` where it landed as this isn't labelled a `release/blocking`. We're not 100% sure this is going to make it into the initial `1.19.x` release and might be pushed into a HF. Once that happens, it will be moved into the correct milestone.

**Comment by kjozwiak** [comment]
Waiting on new `1.19.x` build that will include `88.0.4324.96` 👍

**Comment by GeetaSarvadnya** [comment]
Verification passed on

```

Brave | 1.19.86 Chromium: 88.0.4324.96 (Official Build) (64-bit)
-- | --
Revision | 68dba2d8a0b149a1d3afac56fa74648032bcf46b-refs/branch-heads/4324@{#1784}
OS | Windows 10 OS Version 2004 (Build 19041.746)


```
- Verified the test plan from https://github.com/brave/brave-core/pull/7591
- Verified that all the test cases on the test pages are working as expected
    https://dev-pages.brave.software/referrer/index.html
    https://fmarier.github.io/brave-testing/referrer-spoofing.html
    https://referrer.fmarier.org/redirects.html
    https://referrer.fmarier.org/samesite.html
